bcipy/language/model/gpt2.py

```diff
@@ -23,45 +24,143 @@ def __init__(self, response_type, symbol_set, lm_path=None):
         super().__init__(response_type=response_type, symbol_set=symbol_set)
         self.model = None
         self.tokenizer = None
-        self.is_start_of_word = True
         self.vocab_size = 0
         self.idx_to_word = None
-        self.curr_word_predicted_prob = None
         self.lm_path = lm_path or "gpt2"
+
+        # Hard coding a unigram language model trained on ALS phrase dataset
+        # for smoothing purpose
+        self.unigram_lm = {'E': 0.0998, 'T': 0.096, 'O': 0.0946, 'I': 0.0835,
+                           'A': 0.0774, 'N': 0.0554, SPACE_CHAR: 0.0523, 'H': 0.0504,
+                           'S': 0.0435, 'L': 0.0408, 'R': 0.0387, 'U': 0.0379,
+                           'D': 0.0358, 'Y': 0.0324, 'W': 0.0288, 'M': 0.0266,
+                           'G': 0.0221, 'C': 0.018, 'K': 0.016, 'P': 0.0145,
+                           'F': 0.0117, 'B': 0.0113, 'V': 0.0091, 'J': 0.0016,
+                           'X': 0.0008, 'Z': 0.0005, 'Q': 0.0002, BACKSPACE_CHAR: 0.0}
```

Comment:
I'm not sure the impact of setting a zero on BACKSPACE_CHAR?

Reply:
Currently the language model sets zero probability on BACKSPACE_CHAR, but the copy phrase code would add a nonzero probability to it (I think it is 0.05). I think we can leave it this way for now.